jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/MockHttpStream.java

```diff
@@ -231,6 +231,8 @@ public boolean isCommitted()
     @Override
     public Throwable consumeAvailable()
     {
+        if (_channel.getConnectionMetaData() instanceof MockConnectionMetaData mock)
+            mock.notPersistent();
         return HttpStream.consumeAvailable(this, new HttpConfiguration());
```

Comment:
Don't think this si right. First consumeAvailable(), and only if non-null returned we want to make the connection not persistent.